zebrad/src/components/mempool/storage.rs

```diff
@@ -224,6 +261,9 @@ impl Storage {
         spent_mempool_outpoints: Vec<transparent::OutPoint>,
         height: Option<Height>,
     ) -> Result<UnminedTxId, MempoolError> {
+        // Check that the transaction is standard.
+        self.is_standard_tx(&tx)?;
+
```

Comment:
This should be moved below the check for cached rejections below, because that check is faster.